text/1010-global-vars.rst

```diff
@@ -0,0 +1,179 @@
+::
+
+    Status: Draft
+    Type: Feature
+    Created: 2022-01-15
+    Authors: Elvis Pranskevichus <elvis@edgedb.com>
+    RFC PR: `edgedb/rfcs#0049 <https://github.com/edgedb/rfcs/pull/49>`_
+
+==========================
+RFC 1010: Global variables
+==========================
+
+This RFC proposes adding a mechanism to declare and set global variables
+in a schema and set them on per-session basis.
+
+
+Motivation
+==========
+
+The need for non-local data context frequently exists in programming, and the
+database is no exception.  EdgeDB schemas offer a large variety of ways to
+express the data model with EdgeQL expressions, and the ability to parametrize
+these expressions with contextual per-session data will provide even more
+power.  Consider the ubiquitous concept of authentication, where there is
+a notion of "current user" that is used to filter query results or make other
+decisions.  Currently, such data must be passed as an explicit argument to
+queries and there is no way to pass such context to schema expressions at all.
+
+
+Specification
```

Comment:
Can we mark every statement, the capabilities they use? I.e. `CREATE` and `DROP` are `DDLs` and `SET` and `RESET` are session config.